### ISSUE TITLE
ViLBERT: Pretraining Task-Agnostic Visiolinguistic Representations for Vision-and-Language Tasks

### ISSUE THREAD

**Issue by wwacky** [wwacky/survey#26]
### 論文リンク
- [abstract](https://research.fb.com/publications/vilbert-pretraining-task-agnostic-visiolinguistic-representations-for-vision-and-language-tasks/)
- [pdf](https://research.fb.com/wp-content/uploads/2019/12/ViLBERT-Pretraining-Task-Agnostic-Visiolinguistic-Representations-for-Vision-and-Language-Tasks.pdf?)

### 著者/所属機関
- Jiasen Lu (Georgia Institute of Technology)
- Dhruv Batra (Georgia Institute of Technology, Facebook AI Research)
- Devi Parikh (Georgia Institute of Technology, Facebook AI Research)
- Stefan Lee (Georgia Institute of Technology, Oregon State University)

## 概要

画像と自然言語の2つのモーダルを別々のストリームで処理し、co-attention transformarで相互作用を行わせる。
自動的に集めたConceptual Captions datasetで学習し、基盤に少量の追加を行うことでVision-and-language taskを解かせる。
- vision question answering
- visual commonsense reasoning
- reffering expressions
- caption-based image retrieval

全てのタスクで既存のタスク固有の性能を超えてSOTAを達成した。

## 手法

BERTのself-attentionをマルチモーダル用に修正する。点線ボックス部分は下付きの文字で表されるようにLayerが繰り返される。

![image](https://user-images.githubusercontent.com/4422849/70722406-97c80c00-1d3a-11ea-9477-34fa0b134b67.png)
※ Fig 1. ViLBERT: Pretraining Task-Agnostic Visiolinguistic Representations for Vision-and-Language Tasksより引用

Fig 1のCO-TRM(co-attention transformer layer: Fig2(b))はmulti head attentionのkeyとvalueをVisual StreamとLinguistic Streamで入れ替えて計算を行う。

![image](https://user-images.githubusercontent.com/4422849/70722575-e2498880-1d3a-11ea-85d2-e133c6bd5308.png)
※ Fig 2. ViLBERT: Pretraining Task-Agnostic Visiolinguistic Representations for Vision-and-Language Tasksより引用

### 画像特徴量
- pre-trained object detection networkを用いて物体のバウンディングボックスとその画像特徴量を抽出し、画像領域の特徴量を生成する
    - 実験ではVisual Genome datasetでpretrainしたFaster R-CNN(ResNet-101 が元)を利用
- 単語とは違って抽出された物体領域には適切な順序は存在しないので、空間的な位置を変わりに利用する
    - 位置は正規化した左上と右下の座標、カバーされる画像領域の割合の5次元の合計で表される
    - class detection確率が確信度の閾値を超えているものを選択し、10から36のスコアが高いボックスを残した
    - region iの特徴v_iは領域のmean-pooled convolutional featureで定義される
- Vision streamでは[IMG]特殊トークンを最初にセットする
- visual streamのTransformerとco-attentional transformer blockは1024次元で8 attention headsで行う

### pre-trainで解くタスク
_masked multi modal modeling_と_multi-modal alignment prediction_の2つを行う

#### masked multi modal modeling
- linguistic streamは標準的なBERTのmasked language modelに従う
- 画像はimage regionの90%をゼロベクトルに変更してマスクし、10%はそのまま。マスクされたimage regionに対しては、object detectionで取得されたclass分類を行う
     - 学習時は2値分類ではなく、各クラスの確率分布を正解と予測でKL Divergenceが最小になるようにする(正解はFaster R-CNNから出力されたクラスなので分布にある?)

![image](https://user-images.githubusercontent.com/4422849/71148780-ec561480-226f-11ea-8ef8-bab3c9ae8ce0.png)

#### multi-modal alignment prediction
[IMG]と[CLS]を入力として、画像とキャプションの組み合わせが正しいかの2値分類を行う。Conceptual Captionsデータセットは正しいImage-captionのペアのみなので、Imageかcaptionをランダムに置換してnegativeサンプルを生成する。

![image](https://user-images.githubusercontent.com/4422849/70722661-02794780-1d3b-11ea-96c7-013555ff5f22.png)
※ Fig 3. ViLBERT: Pretraining Task-Agnostic Visiolinguistic Representations for Vision-and-Language Tasksより引用

![image](https://user-images.githubusercontent.com/4422849/71149251-57541b00-2271-11ea-9b36-ae25467883bf.png)
![image](https://user-images.githubusercontent.com/4422849/71149262-63d87380-2271-11ea-9b13-500e7b5d6ff5.png)

### データ
- Conceptual Captionsは330万のimage-captionのペアで、Web imageからalt-textをスクレイピングして自動的に作成
- データの自動収集と下処理ではノイズや人間らしくなかったり、短すぎるものもそのまま
- ダウンロード時にいくつかのリンクは破損しているので、学習は310万のimage-captionのペアで実施

### pretraining
- linguistic stream
    - linguistic streamはBookCorpusとEnglish Wikipediaでpretrainしたモデルを初期値
    - BERT_BASEの12層、762次元、12attention headsの構造を使用

- training setting
    - 8 Titan X GPU上でバッチサイズ512で10エポック実行
    - 初期学習率1e-4でADAMで最適化
    - linear decay learning rate schedule with warm upを使用した
    - 両学習タスクのlossのウェイトは等しくした

#### Visual Question Answering (VQA)
- 画像について自然言語で回答を行うタスク
- VQA 2.0 datasetを使用
    - COCOの画像に対して110万の質問があり、それぞれに10個の正解がある

#### Visual Commonsense Reasoning (VCR)
- 画像がgivenで、2つの問題を解く。
    - visual question answering(V→A)とanswer justification(QA→R)があり、両方とも当たっているかどうかで判定 
- VCRデータセットは11万の映画シーンから作成された29万のQAの問題がある

![image](https://user-images.githubusercontent.com/4422849/71146088-253dbb80-2267-11ea-9f68-e4d5e23aab6a.png)
※※ https://visualcommonsense.com/より引用

#### Grounding Referring Expressions (RefCOCO+ データセット)

![image](https://user-images.githubusercontent.com/4422849/71145978-c9733280-2266-11ea-8b7b-bfa05aab1bd5.png)
※ Grounding Referring Expressions in Images by Variational Contextより引用

#### Caption-Based Image Retrieval (Image Retrieval)
- captionのテキストがgivenで、プールから該当する画像を探すタスク
- Flickrからそれぞれに5つcaptionがついた31,000画像を集めたものをデータセット:Flickr30kとした。この内の1,000画像をvalidationに使用する

#### 'Zero-shot' Caption-Based Image Retrieval (ZS Image Retrieval)

fine-tuningなしでCaption-Based Image Retrievalを行ったもの

## 結果

Baselineに2つのモデルを用いる

**Single-Stream**
- Transformer blockをVisionとLanguageで分けずに1ラインでパラメータを共有する手法を試す。同じTransofomer blockにVision・Language両方を入力する。  
- BERT BASEを初期値としてVilBERTと同様に学習させる
- VilBERT(提案手法)の方はVision Languageで分かれているのでtwo-stream
- 計算コストが高すぎるため、Image RetrievalとZero-shot image retrievalでは評価を行っていない。

**VilBERT†**
- pretrainを行っていないVilBERT構造のネットワーク
- linguistic streamはBERTの初期値、image regionの分散表現はfull VilBERTと同じFaster R-CNNを用いる
- pretrain無しでtask-specific baseline modelからのgainがどの程度あるか比較するために行う。(Conceptual Captionのpretrainingの効果を見る)

base lineと提案手法はfinetuneを行う。

**Task-Specific Baselines**
各タスクのSOTAの手法を用いる。

### Results

- two-stream modelはsingle-stream modelよりも性能が良い
    - pretrain有り(Single-Stream vs VilBERT)でも、pretrain無し(Single-Stream† vs VilBERT†)でも両方ともViLBERT構造(two-stream model)の方が性能が良い。
- pretraining taskはvisionlinguistic分散表現より性能が良い
    - (ViLBERT vs ViLBERT†)の比較で分かるようにpretrainで性能が良くなる
- task-specific modelよりViLBERT(finetuneあり)の方が性能が良い

![image](https://user-images.githubusercontent.com/4422849/70722780-39e7f400-1d3b-11ea-9e5f-804ec00393b6.png)
※ Table 1. ViLBERT: Pretraining Task-Agnostic Visiolinguistic Representations for Vision-and-Language Tasksより引用
ーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーー

**Vision Streamの効果**
Table 2にCO-TRM→TRM block(Fig 1の点線部分)の数による性能の変化を記載。  
Vision Streamはembeddingの後にCO-TRM→TRM blockが来るので、Vision Streamの層の数を調整していることになる。BERTはCO-TRM→TRM blockの前に一度Transformerを挟んでいる。
- VQA、Image Retrieval(Zero-shot Image Retrieval含む？)では、の数を増やすと性能が単調に増加する
- VCR、RefCOCO+では階層が少ないほうが性能が良い

![image](https://user-images.githubusercontent.com/4422849/70722907-6865cf00-1d3b-11ea-95f8-2d515b5747ba.png)
※ Table 2. ViLBERT: Pretraining Task-Agnostic Visiolinguistic Representations for Vision-and-Language Tasksより引用
  
ーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーーー
  
**Benefits of Large Training Sets**
Table 3にpre-trainingに利用するconceptual caption dataset を何割利用するか変更した結果を記載。pre-trainingとfinetuneの両方を行う。  
pre-trainingに利用するデータ量を増やすと、精度は単調に増加する。

![image](https://user-images.githubusercontent.com/4422849/70722984-83384380-1d3b-11ea-8fe5-a2de37415f27.png)
※ Table 3. ViLBERT: Pretraining Task-Agnostic Visiolinguistic Representations for Vision-and-Language Tasksより引用

## 関連研究
VideoBERT
- 動画と自然言語のjoint representationを行っている
- 構造としては本研究(VilBERT)のSingle streamに相当

LXMERT
- cross-modality modelでVilBERTより独自の構成をしている
- pre-trainingのデータにはCOCOとVisual Genomeを利用している

他の見込みのあるjoint-linguistic pretrainingには以下が挙げられる
- VLBERT
- Unified VLP
- UNITER

## 略語
頭字語 | 意味
--- | ---
ViLBERT | Vision-and-Language BERT　
TRM | Transformer blocks
Co-TRM | co-attentional transformer layers
